[Chan Kok Jing, Daryl] Duke Increments

dvrylc opened the pull request.

j-lum committed:
Add support for Gradle workflow

j-lum committed:
Add sample checkstyle configuration

Add toolVersion block in to Gradle code sample to prevent errors.

dvrylc committed:
Merge branch 'gradle'

dvrylc committed:
Make gradlew executable

dvrylc committed:
Configure gradle

- Add checkstyle and shadow plugins
- Add JUnit dependency and test folder

dvrylc committed:
Merge pull request #1 from dvrylc/configure-gradle

Add and configure gradle

dvrylc committed:
[Level 1] Add basic scaffold

dvrylc committed:
[Level 2] Implement add and list features

dvrylc committed:
[Level 3] Add ability to mark tasks as done

dvrylc committed:
[Level 4] Add todo, deadline and event classes

dvrylc committed:
[Level 4] Update base task class

dvrylc committed:
[Level 4] Add support for new types of tasks

dvrylc committed:
[A-TextUiTesting] Add test script and data

dvrylc committed:
[Level 5] Add error handling using exceptions

dvrylc committed:
[Level 6] Implement deletion of tasks

dvrylc committed:
Merge pull request #2 from dvrylc/week2

Complete week 2 tasks

dvrylc committed:
Fix checkstyle

- Add Javadoc for main method
- Fix `case` indentation

dvrylc committed:
[Level-7] Add save format to tasks

dvrylc committed:
[Level-7] Add overloaded constructors to tasks

dvrylc committed:
[Level-7] Implement saving to filesystem

dvrylc committed:
[Level-8] Implement dates and times for events and deadlines

dvrylc committed:
Merge branch 'branch-Level-7'

dvrylc committed:
Merge branch 'branch-Level-8'

dvrylc committed:
Refactor date formatter into separate utility class

dvrylc committed:
Fix date parser for save file

dvrylc committed:
[A-MoreOOP] Overall refactor

dvrylc committed:
[A-JUnit] Add basic unit testing

dvrylc committed:
[A-JavaDoc] Add Javadocs

dvrylc committed:
[Level-9] Add find feature

dvrylc committed:
Merge branch 'branch-A-JavaDoc'

dvrylc committed:
Merge branch 'branch-Level-9'

dvrylc committed:
Implement duke package

dvrylc committed:
Refactor tests

dvrylc committed:
Overall improvements

- Change test method names to comply with the coding standard
- Package classes

yan-wl commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,50 @@
+
+import java.util.Date;
+
+class Deadline extends Task {
```
Nice

yan-wl commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,110 @@
+
+public class Duke {
+    // Configuration
+    static final SimpleDateFormat dateFormatter = new SimpleDateFormat("dd/MM/yyyy HHmm");
```
I believe this identifier should be all caps as it is a constant

yan-wl commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,110 @@
+                    break;
+                }
+
+                case "done":
```
Needs a _//fallthrough_ to comply with coding standards/style

yan-wl commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,106 @@
+    }
+
+    /**
+     * Generates a numbered list of tasks matching a query.
```
This description does not seem to tally with what the method is doing

dvrylc replied:
Hmm, could you elaborate on this please?

yan-wl replied:
Ah, I was mistaken. The return description gave me the impression that it would return a List object instead of the String representation of a list.

yan-wl reviewed (commented):
Good job overall!

dvrylc committed:
[Level-10] Add GUI for Duke

dvrylc committed:
Merge branch 'branch-Level-10'

dvrylc committed:
Add GUI assets and update gitignore

dvrylc committed:
Fix issues raised during peer review

- Fix constant name to comply with coding standard
- Fix `case` missing fallthrough comment

dvrylc committed:
[A-Assertions] Add assertions

dvrylc committed:
Improve code quality (#8)

* [A-CodeQuality] Add missing Javadocs
* [A-CodeQuality] Refactor file storage

dvrylc committed:
[A-Travis] Add travis config

dvrylc committed:
Fix `assert` checkstyle errors (#9)

dvrylc committed:
[C-FlexibleDataSource] Add support for switching save files

dvrylc committed:
Merge branch 'branch-C-FlexibleDataSource'

dvrylc committed:
Improve gradle config and handling of missing save file

dvrylc committed:
Overall refactor for better OOP

dvrylc committed:
Improve UI

dvrylc committed:
[A-UserGuide] Add user guide and screenshot

dvrylc committed:
Merge branch 'branch-A-UserGuide'

dvrylc committed:
Set theme jekyll-theme-minimal

dvrylc committed:
Add missing JavaDoc to fix checkstyle
